frontend/src/Components/roadDetails/TopBar.tsx

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import { Link, useNavigate } from 'react-router-dom';
+
+interface TopBarProps {
+  setShowMiddleArea2: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+const return_btn = (
+  <svg
+    xmlns="http://www.w3.org/2000/svg"
+    width="16"
+    height="16"
+    fill="currentColor"
+    className="btn bi bi-chevron-left btnWhite"
+    viewBox="0 0 16 16"
+  >
+    <path
+      fillRule="evenodd"
+      d="M11.354 1.646a.5.5 0 0 1 0 .708L5.707 8l5.647 5.646a.5.5 0 0 1-.708.708l-6-6a.5.5 0 0 1 0-.708l6-6a.5.5 0 0 1 .708 0z"
+    />
+  </svg>
+);
+
+const TopBar: React.FC<TopBarProps> = ({ setShowMiddleArea2 }) => {
```

Comment:
Add documentation, for instance:
```suggestion
/**
 * A bar at the top of the page with a button to go back to the root page of the website and a toggle button
 */
const TopBar: React.FC<TopBarProps> = ({ setShowMiddleArea2 }) => {
```